File: lib/twilio.go
package lib

import (
	"context"
	"fmt"
	"os"

	"github.com/twilio/twilio-go"
	twilioApi "github.com/twilio/twilio-go/rest/api/v2010"
)

var twilioClient *twilio.RestClient

// InitTwilio initializes the Twilio client
func InitTwilio() error {
	accountSID := os.Getenv("TWILIO_ACCOUNT_SID")
	authToken := os.Getenv("TWILIO_AUTH_TOKEN")

	if accountSID == "" || authToken == "" {
		return fmt.Errorf("TWILIO_ACCOUNT_SID or TWILIO_AUTH_TOKEN not set")
	}

	twilioClient = twilio.NewRestClientWithParams(twilio.ClientParams{
		Username: accountSID,
		Password: authToken,
	})

	return nil
}

// SendWhatsAppMessage sends a WhatsApp message via Twilio
func SendWhatsAppMessage(ctx context.Context, to, message string) error {
	whatsappNumber := os.Getenv("TWILIO_WHATSAPP_NUMBER")
	if whatsappNumber == "" {
		return fmt.Errorf("TWILIO_WHATSAPP_NUMBER not set")
	}

	params := &twilioApi.CreateMessageParams{}
	params.SetTo(to)
	params.SetFrom(whatsappNumber)
	params.SetBody(message)

	resp, err := twilioClient.Api.CreateMessage(params)
	if err != nil {
		return fmt.Errorf("failed to send WhatsApp message: %w", err)
	}

	if resp.Sid == nil {
		return fmt.Errorf("no message SID returned from Twilio")
	}

	return nil
}

// GetTwilioClient returns the Twilio client instance
func GetTwilioClient() *twilio.RestClient {
	return twilioClient
}

File: lib/twilio.ts
import twilio from 'twilio';

const accountSid = process.env.TWILIO_ACCOUNT_SID;
const authToken = process.env.TWILIO_AUTH_TOKEN;
const fromNumber = process.env.TWILIO_WHATSAPP_NUMBER;

if (!accountSid || !authToken) {
  console.error('Missing Twilio credentials');
}

export const twilioClient = twilio(accountSid, authToken);

export async function sendWhatsAppMessage(to: string, body: string) {
  if (!fromNumber) {
    throw new Error('TWILIO_WHATSAPP_NUMBER is not set');
  }

  try {
    const message = await twilioClient.messages.create({
      body: body,
      from: fromNumber,
      to: to,
    });
    console.log(`Message sent to ${to}: ${message.sid}`);
    return message;
  } catch (error) {
    console.error('Error sending WhatsApp message:', error);
    throw error;
  }
}

